src/Form/FormAnalyzer.js

```diff
@@ -209,8 +209,8 @@ class FormAnalyzer {
 
         // A form with many fields is unlikely to be a login form
         const relevantFields = this.form.querySelectorAll(this.matching.cssSelector('GENERIC_TEXT_FIELD'))
-        if (relevantFields.length > 3) {
-            this.increaseSignalBy(2, 'many fields: it is probably not a login')
+        if (relevantFields.length > 4) {
+            this.increaseSignalBy(relevantFields.length * 1.5, 'many fields: it is probably not a login')
```

Comment:
This gives more relevance to the fact that forms with several fields are more likely to be signup/checkout forms than login. Instead of giving a fixed value, the value increases more than proportionally to the number of fields. This will have a significant impact on accuracy of checkout forms where several fields are the norm.